File: Checkers/Assets/Scripts/Deplacement.cs
using UnityEngine;

public class Deplacement
{
    public Vector2Int Origin { get; }
    public Vector2Int Destination { get; }

    public Deplacement(int x1, int y1, int x2, int y2)
    {
        Origin = new Vector2Int(x1, y1);
        Destination = new Vector2Int(x2, y2);
    }

    public bool NullMove()
    {
        return Origin == Destination;
    }

    // Returns the position of the eaten piece, if there is no, the return is a Vector2Int.zero => (0, 0)
    public Vector2Int EatenPiece()
    {
        int sumX = Origin.x + Destination.x;
        int sumY = Origin.y + Destination.y;
        bool okX = (sumX) % 2 == 0;
        bool okY = (sumY) % 2 == 0;
        return okX && okY ? new Vector2Int(sumX / 2, sumY / 2) : Vector2Int.zero;
    }
}

File: CheckersPythonEngine/game_engine/Deplacement.py
class Deplacement:
    origin: tuple
    destination: tuple

    def __init__(self, x1: int, y1: int, x2: int, y2: int):
        self.origin = (x1, y1)
        self.destination = (x2, y2)

    def eaten_piece(self):
        """

        :return: Returns the position of the eaten piece, if there is no, the return is None.
        """
        sum_x = self.origin[0] + self.destination[0]
        sum_y = self.origin[1] + self.destination[1]
        ok_x = sum_x % 2 == 0
        ok_y = sum_y % 2 == 0
        return (sum_x / 2, sum_y / 2) if ok_x and ok_y else None

    def null_move(self):
        return self.origin == self.destination
